astropy/modeling/core.py

```diff
@@ -4492,7 +4458,7 @@ def _custom_model_inputs(func):
             special_params[param.name] = param.default
         elif param.name in settable:
             settable_params[param.name] = param.default
-        elif param.name in properties:
+        elif param.name in properties or param.name in reserved:
```

Comment:
this is to preserve exact backward compatibility as we have a test checking for this specific use case. Maybe the correct fix should be to add `"n_inputs"` to `special` and update the test instead. Maybe this change would make more sense in a dedicated PR.

Reply:
moved to #16633